utils/skyportal_api.py

```diff
@@ -0,0 +1,1090 @@
+    Update a classification to skyportal using its API
+
+    Parameters
+    ----------
+    object_id : str
+        Id of the object for which we update the classification
+    classification : str
+        Classification of the object
+    probability : float
+        Probability of classification
+    taxonomy_id : int
+        id of the taxonomy in which the classification is defined
+    group_ids : list
+        List of group ids to post the classification to
+    url : str
+        Skyportal url
+    token : str
+        Skyportal token
+
+    Returns
+    ----------
+    status_code : int
+        HTTP status code
+    """
+
+    data_classification = classification_id_for_objs(object_id, url, token)[1]
+    classification_id, author_id = (
+        data_classification["id"],
+        data_classification["author_id"],
+    )
+
+    data = {
+        "obj_id": object_id,
+        "classification": classification,
+        "probability": probability,
+        "taxonomy_id": taxonomy_id,
+        "group_ids": group_ids,
+        "author_id": author_id,
+        "author_name": "fink_client",
+    }
+
+    response = api(
+        "PUT",
+        f"{url}/api/classification/{classification_id}",
+        data,
+        token=token,
+    )
+    return response.status_code
+
+
+def get_all_filters(url: str, token: str):
+    """
+    Get all filters from skyportal using its API
+
+    Parameters
+    ----------
+    url : str
+        Skyportal url
+    token : str
+        Skyportal token
+
+    Returns
+    ----------
+    status_code : int
+        HTTP status code
+    data : list
+        List of filters
+    """
+    response = api("GET", f"{url}/api/filters", token=token)
+    return (response.status_code, response.json()["data"])
+
+
+def get_all_taxonomies(url: str, token: str):
+    """
+    Get all taxonomies from skyportal using its API
+
+    Parameters
+    ----------
+    url : str
+        Skyportal url
+    token : str
+        Skyportal token
+
+    Returns
+    ----------
+    status_code : int
+        HTTP status code
+    data : list
+        List of taxonomies
+    """
+    response = api("GET", f"{url}/api/taxonomy", token=token)
+    return (response.status_code, response.json()["data"])
+
+
+def init_skyportal(url: str, token: str):
+    """
+    Creates the different entities needed in skyportal to add the data of alerts from fink
+    (streams, filters, groups) and returns the ids of the entities created, so they can be used to post the alerts to skyportal using its API
+
+    Parameters
+    ----------
+    url : str
+        Skyportal url
+    token : str
+        Skyportal token
+
+    Returns
+    ----------
+    fink_id : int
+        Id of the fink group in SkyPortal
+    stream_id : int
+        Id of the fink stream in SkyPortal
+    filter_id : int
+        Id of the fink filter in SkyPortal
+    """
+    streams = get_all_streams(url, token)[1]
+    # check if a stream with name 'fink_stream' exists
```

Comment:
Just to be sure I understand -- 'fink_stream' is an internal of skyportal?

Reply:
In SkyPortal's DB, for different reasons, there is a "streams" table, and it is necessary that some elements we add to the DB are linked to a stream, which is why we create a stream that corresponds to fink so we can reference it when adding photometries and such later on.